esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2940,58 +2956,45 @@ bool CFileSprayEx::onDropZoneFileSearch(IEspContext &context, IEspDropZoneFileSe
-        queryDropZoneInfo(context.getClientVersion(), dropZone, req.getECLWatchVisibleOnly(), dropZones);
-        if (dropZones.empty())
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s not found.", dropZone);
+        Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory();
+        Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
+        Owned<IConstDropZoneInfo> dropZoneInfo = constEnv->getDropZone(dropZoneName);
+        if (!dropZoneInfo || (req.getECLWatchVisibleOnly() && !dropZoneInfo->isECLWatchVisible()))
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s not found.", dropZoneName);
 
-        IArrayOf<IEspPhysicalFileStruct> files;
-        IConstTpDropZone& dropZoneInfo = dropZones.item(0);
-        IArrayOf<IConstTpMachine>& dropZoneServers = dropZoneInfo.getTpMachines();
-        if (dropZoneServers.empty())
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "No server found for dropzone %s.", dropZone);
+        SCMStringBuffer directory, computer;
+        dropZoneInfo->getDirectory(directory);
+        if (!directory.length())
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone Directory not found for %s.", dropZoneName);
 
-        if (isEmptyString(dropZoneServers.item(0).getType()))
+        IpAddress ipToMatch;
+        const char* dropZoneServerReq = req.getServer(); //IP or hostname
+        if (!isEmptyString(dropZoneServerReq))
         {
-            const char* dropZoneServer = req.getServer();
-            IpAddress ipToMatch;
-            if (!isEmptyString(dropZoneServer))
-            {
-                ipToMatch.ipset(dropZoneServer);
-                if (ipToMatch.isNull())
-                    throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid server %s found.", dropZoneServer);
-            }
-
-            ForEachItemIn(i, dropZoneServers)
-            {
-                IConstTpMachine& server = dropZoneServers.item(i);
-                const char* address = server.getNetaddress();
-                if (isEmptyString(address))
-                    continue;
-
-                IpAddress ip(address);
-                if (isEmptyString(dropZoneServer))
-                    appendDropZoneFiles(context, ip, server.getDirectory(), req.getNameFilter(), files);
-                else if (ip.ipequals(ipToMatch))
-                {
-                    appendDropZoneFiles(context, ip, server.getDirectory(), req.getNameFilter(), files);
-                    break;
-                }
-            }
+            ipToMatch.ipset(dropZoneServerReq);
+            if (ipToMatch.isNull())
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid server %s specified.", dropZoneServerReq);
         }
-        else //legacy DZ settings in environment.xml
+
+        IArrayOf<IEspPhysicalFileStruct> files;
+        Owned<IConstDropZoneServerInfoIterator> machineItr = dropZoneInfo->getServers();
```

Comment:
The code below is another case where it could be cleaner/simpler if you encapsulated a getDropZonesByIP() iterator and used it.